client/client-lib/src/lib.rs

```diff
@@ -111,6 +112,70 @@ impl From<GatewayClientConfig> for LightningGateway {
     }
 }
 
+#[derive(Clone, Debug, Encodable, Decodable, Serialize, Deserialize)]
+pub enum PaymentDirection {
+    Inbound,
+    Outbound,
+}
+
+#[derive(Clone, Debug, Encodable, Decodable, Serialize, Deserialize)]
+pub enum TypedPaymentDetail {
+    InboundOnchain {
+        tx_id: bitcoin::Txid,
+        confirmed_in_block: Option<bitcoin::BlockHash>,
+        proof: PegInProof,
+    },
+    OutboundOnchain {
+        tx_id: bitcoin::Txid,
+        confirmed_in_block: Option<bitcoin::BlockHash>,
+    },
+    InboundLightning {
+        invoice: Invoice,
+        paid: bool,
+        used_gateway: PublicKey,
+    },
+    OutboundLightning {
+        used_invoice: Invoice,
+        contract_id: ContractId,
+        used_gateway: PublicKey,
+    },
+    InboundMint {
+        used_token: TieredMulti<SpendableNote>,
+        reissued_token: Option<TieredMulti<SpendableNote>>,
+    },
+    OutboundMint {
+        token: TieredMulti<SpendableNote>,
+    },
+}
+
+#[derive(Clone, Debug, Encodable, Decodable, Serialize, Deserialize)]
+pub struct Payment {
+    pub uuid: Uuid,
+    pub timestamp: u128,
+    pub amount: Amount,
+    pub direction: PaymentDirection,
```

Comment:
The direction is already included in the `detail` field imo, couldn't `TypedPaymentDetail` have a function `direction(&self) -> PaymentDirection` instead?